packages/tooling/fast-tooling-react/src/form/custom-controls/control.css.md

```diff
@@ -0,0 +1,140 @@
+# Abstract
+
+The CSS controls consist of an abstract control and two extended controls, inline CSS and internal CSS. These controls will provide a way to interpret CSS strings for inline and internal CSS into form UI and back to strings.
+
+## Build script
+
+The package should have a `devDependency` for [MDN data](https://www.npmjs.com/package/mdn-data) and re-interpret the properties based on this. To only require what is needed there should be a build script which takes the MDN data and converts it to a `.ts` file with exports conforming to an interface which defines the example provided in [Variables](#variables).
+
+## Variables
+
+There should be a list of internal variables that are assigned to each CSS property. The structure should look like this:
+
+```typescript
+[
+    {
+        "border": { // The property name should be pascalCase
+            name: "border", // The name should be spinal-case
+            shorthand: true, // denotes that "border" CSS property is a shorthand
+            default: "none",
+            globalValues: [
+                {
+                    value: "inherit"
+                },
+                {
+                    value: "initial"
+                },
+                {
+                    value: "unset"
+                }
+            ],
+            values: [ // these must be in order
+                {
+                    mapsToCssProperty: "border-width",
+                    type: "width",
+                    default: "1px"
+                },
+                {
+                    mapsToCssProperty: "border-style",
+                    type: "border-style",
+                    default: "solid"
+                },
+                {
+                    mapsToCssProperty: "border-color",
+                    type: "color",
+                    default: "black"
+                }
+            ]
+        }
+    },
+    // Complete list of all potential CSS properties
+]
+```
+
+From this structure we can construct via the UI generated from the below methods: `border: 1px solid red`.
+
+## Methods
+
+### Render methods
+
+Render methods should be created for the following inputs:
+- Text field
+    - Used as a default value if the property name is not recognized
+- Autocomplete dropdown
+    - Used by the property name selector, once this has been chosen, the value should reflect different inputs
+    - Used by color (full list of available colors can be autocompleted)
+    - Used by border-style
+    - Used by anything which could use a CSS function such as var()
```

Comment:
Can you expand more on how things like `var` and `calc` would be handled?

Reply:
Now that I've thought about it, CSS functions need their own supplied input, I'll re-work this section.